modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/utils/AlertUtils.scala

```diff
@@ -0,0 +1,82 @@
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.snowplow.rdbloader.utils
+
+import cats.{Applicative, Monad}
+import cats.implicits._
+import com.snowplowanalytics.iglu.core.SelfDescribingData
+import com.snowplowanalytics.iglu.core.circe.implicits.igluNormalizeDataJson
+import com.snowplowanalytics.snowplow.rdbloader.LoaderAction
+import com.snowplowanalytics.snowplow.rdbloader.common.S3
+import com.snowplowanalytics.snowplow.rdbloader.common.S3.{Folder, Key}
+import com.snowplowanalytics.snowplow.rdbloader.db.Statement.{CreateTempTable, InsertIntoTempTable, Minus}
+import com.snowplowanalytics.snowplow.rdbloader.dsl.Monitoring.{AlertPayload, AlertSchemaKey}
+import com.snowplowanalytics.snowplow.rdbloader.dsl.{AWS, JDBC, Monitoring}
+import com.snowplowanalytics.snowplow.rdbloader.generated.BuildInfo
+import io.circe.Json
+import io.circe.syntax.EncoderOps
+
+
+object AlertUtils {
+
+  def listAllBatches[F[_]: Applicative](aws: AWS[F], shreddedPath: S3.Folder): LoaderAction[F, List[S3.BlobObject]] =
+    LoaderAction.liftF(aws.listS3(shreddedPath))
+
+  def createTempTable[F[_]: JDBC](tempTableName: String): LoaderAction[F, Int] =
+    JDBC[F].executeUpdate(CreateTempTable(tempTableName))
+
+  def insertAllBatchesToTemp[F[_]: JDBC](tempTableName: String, objects: List[S3.BlobObject]): LoaderAction[F, Int] =
+    JDBC[F].executeUpdate(InsertIntoTempTable(tempTableName, objects.map(_.key)))
+
+  def tempMinusManifest[F[_]: JDBC](tempTableName: String, manifestTableName: String): LoaderAction[F, List[String]] =
+    JDBC[F].executeQueryList(Minus(tempTableName, manifestTableName))
+
+  def checkShreddingComplete[F[_]: Applicative](aws: AWS[F], keys: List[String]): LoaderAction[F, List[(String, Boolean)]] =
+    LoaderAction.liftF[F, List[(String, Boolean)]] {
+      keys
+        .map { key =>
+          aws.keyExists(Key.coerce(key + "/shredding_complete.json")).map(b => (key, b))
+        }
+        .sequence
+    }
+
+  def postPayload(payload: AlertPayload): String =
+    SelfDescribingData[Json](AlertSchemaKey, payload.asJson).normalize.noSpaces
+
+  def createAlertPayload(folder: Folder, message: String, tags: Map[String, String]): AlertPayload =
+    AlertPayload(BuildInfo.version, folder.folderName , message, tags)
+
+  def createAlertPayloads[F[_]: JDBC: Monad](
+    aws: AWS[F],
```

Comment:
I think `AWS` is a capability trait, so could be pased as `def createAlertPayloads[F[_]: AWS`